web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -132,60 +122,52 @@ export class LayerDialogComponent implements OnDestroy {
     this.layerName = this.layer?.name?.get(this.lang) || '';
     this.color = this.layer?.color || DEFAULT_LAYER_COLOR;
     if (!layer) {
-      this.layer = this.createNewLayer();
-      const newField = this.createNewField();
+      this.layer = this.layerService.createNewLayer();
+      const newField = this.layerService.createField(
+        FieldType.TEXT,
+        /* label= */
+        '',
+        /* required= */
+        false,
+        /* index= */
+        this.fields.size
+      );
       this.fields = this.fields.push(newField);
       return;
     }
-    const form = this.getForms();
+    const form = this.layerService.getForm(this.layer);
     if (form) {
       this.fields =
-        this.getForms()
-          ?.fields.toList()
-          .sortBy(field => field.index) || List<Field>();
+        form?.fields.toList().sortBy(field => field.index) || List<Field>();
     }
   }
 
-  private getForms(): Form | undefined {
-    const forms = this.layer?.forms;
-    return forms ? forms.valueSeq().first() : undefined;
-  }
-
   onSave() {
     // TODO: Wait for project to load before showing dialog.
     if (!this.projectId) {
       throw Error('Project not yet loaded');
     }
-    let fields = Map<string, Field>();
     // Check if there are empty fields, if empty return.
     const emptyFields = this.fields.filter(field => !field.label.get('en'));
     if (emptyFields.size) {
       return;
     }
-    this.fields.forEach((field: Field, index: number) => {
-      const layerFieldId = this.fields && this.fields.get(index)?.id;
-      const fieldId = layerFieldId
-        ? layerFieldId
-        : this.dataStoreService.generateId();
-      fields = fields.set(fieldId, field);
-    });
-    const form = this.getForms();
-    const formId = form ? form.id : this.dataStoreService.generateId();
+    const fields = this.layerService.convertFieldsListToMap(this.fields);
+    const form = this.layerService.getForm(this.layer);
+    const forms = form
```

Comment:
I think the current logic could be further simplified - instead of extracting the form id from the layer using the service, can we add it as a member to this component? Then we wouldn't need to merge the previous form with the new fields; the component will have id and fields that could be passed to the LayerService to create or update the form.